realm/realm-library/src/main/java/io/realm/RealmSchema.java

```diff
@@ -34,324 +31,82 @@
  * <p>
  * All changes must happen inside a write transaction for the particular Realm.
  *
- * @see io.realm.RealmMigration
+ * @see RealmMigration
  */
-public class RealmSchema {
+public abstract class RealmSchema {
+    private ColumnIndices columnIndices; // Cached field look up
 
-    private static final String TABLE_PREFIX = Table.TABLE_PREFIX;
-    private static final String EMPTY_STRING_MSG = "Null or empty class names are not allowed";
-
-    // Caches Dynamic Class objects given as Strings to Realm Tables
-    private final Map<String, Table> dynamicClassToTable = new HashMap<String, Table>();
-    // Caches Class objects (both model classes and proxy classes) to Realm Tables
-    private final Map<Class<? extends RealmModel>, Table> classToTable = new HashMap<Class<? extends RealmModel>, Table>();
-    // Caches Class objects (both model classes and proxy classes) to their Schema object
-    private final Map<Class<? extends RealmModel>, RealmObjectSchema> classToSchema = new HashMap<Class<? extends RealmModel>, RealmObjectSchema>();
-    // Caches Class Strings to their Schema object
-    private final Map<String, RealmObjectSchema> dynamicClassToSchema = new HashMap<String, RealmObjectSchema>();
-
-    private final BaseRealm realm;
-    private long nativePtr;
-    ColumnIndices columnIndices; // Cached field look up
-
-    /**
-     * Creates a wrapper to easily manipulate the current schema of a Realm.
-     */
-    RealmSchema(BaseRealm realm) {
-        this.realm = realm;
-        this.nativePtr = 0;
-    }
-
-    /**
-     * Creates a wrappor to easily manipulate Object Store schemas. This constructor should only be called by
-     * proxy classes during validation of schema.
-     */
-    RealmSchema() {
-        // This is the case where the schema is created from the proxy classes.
-        // dynamicClassToSchema is used to keep track of which model classes have been processed.
-        this.realm = null;
-        this.nativePtr = 0;
-        // TODO: create a Object Store realm::Schema object and store the native pointer
-    }
-
-
-    RealmSchema(ArrayList<RealmObjectSchema> realmObjectSchemas) {
-        long list[] = new long[realmObjectSchemas.size()];
-        for (int i = 0; i < realmObjectSchemas.size(); i++) {
-            list[i] = realmObjectSchemas.get(i).getNativePtr();
-        }
-        this.nativePtr = nativeCreateFromList(list);
-        this.realm = null;
-    }
-
-    public long getNativePtr() {
-        return this.nativePtr;
-    }
-
-    public void close() {
-        if (nativePtr != 0) {
-            Set<RealmObjectSchema> schemas = getAll();
-            for (RealmObjectSchema schema : schemas) {
-                schema.close();
-            }
-            nativeClose(nativePtr);
-        }
-    }
+    public abstract void close();
```

Comment:
Since all these methods are what users will see, it is important they have proper Javadoc. It is actually less important for the actual implementation classes.